src/app/beer_garden/db/mongo/pruner.py

```diff
@@ -30,7 +32,17 @@ def add_task(
             }
         )
 
+    def cancel_in_progress(self):
+        in_progress = Request.objects(Q2(status="IN_PROGRESS") | Q2(status="CREATED"))
+        current_time = datetime.utcnow()
+        for request in in_progress:
+            timeout = request.created_at + timedelta(minutes=self._ttl)
+            if current_time >= timeout:
```

Comment:
You should be able to tack on the timeout check as criteria when you're querying for the Requests.  That is, rather than getting **all** "IN_PROGRESS" and "CREATED" requests and then checking them in application space to determine if they've timed out, just query for the ones that have timed out.  Then you could iterate over them, or possibly just do a single bulk update.